File: 2367-number-of-arithmetic-triplets/2367-number-of-arithmetic-triplets.py
class Solution:
    def arithmeticTriplets(self, nums: List[int], diff: int) -> int:
        

        nl = len(nums) 
        ans = 0 
        for i in range(nl):
            for j in range(i+1, nl):
                for k in range(j+1, nl): 
                    if nums[j]-nums[i] == nums[k]- nums[j] == diff:
                        ans +=1 
        return ans 

File: 2367-number-of-arithmetic-triplets/2367-number-of-arithmetic-triplets.java
class Solution {
    public int arithmeticTriplets(int[] nums, int diff) {
        
        
        Set<Integer> hash = new HashSet<>(); 
        int ans = 0 ; 
        for(int n:nums){
            if(hash.contains(n-diff) && hash.contains(n-2*diff)) ans ++ ;
        
        hash.add(n) ;
        }
        return ans ; 
        
    }
}